### PR TITLE
NewDividerWidget

### LINKED ISSUE #198: add seperator widget
Add a vertical spacer element with an optional title.

**acceptance criteria**
* vertically seperats groups of widgets (e.g. `hr`)
* can display an optional title

### PR DESCRIPTION
**Pull request by RudiKlassen**
closes #198 -> add seperator widget , thx @MatthiasBalke for your Tutorial :)

### CODE REVIEW

**Comment by RudiKlassen** [comment]
unbenannt-1

The new Widget displays a title, a divider and an optional description text

**Comment by matthiasbalke** [comment]
@RudiKlassen please check the error message of your failed build by clicking on Details, next to the required circleci check. After checking what went wrong please fix it and push the changes.

**Comment by matthiasbalke** [comment]
@RudiKlassen also please update your branch to the latest upstream/master and resolve the conflict.

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div class="col-md-12">
+  <h2>{{title}}</h2>
+  <h5>&nbsp;&nbsp;&nbsp;{{description}}</h5>
```
Comment (matthiasbalke) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html]: How about adding a `border-bottom` in combination with some `padding` to the `h2` tag instead of using a `h2` and a `hr`.  This will save some valueable space.

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<div class="col-md-12">
+  <h2>{{title}}</h2>
```
Comment (matthiasbalke) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html]: IMHO you should use a smaller font-size for the heading here. Your `h2` is even bigger than the `h1` title of the dashboard itself.

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html (hunk 1 above)
Comment (matthiasbalke) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html]: Use CSS property `margin-left` on the `h5` tag to indent this heading, instead of using non breaking spaces.

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  templateUrl: './divider-widget.component.html',
+  styleUrls: ['./divider-widget.component.css']
+})
+export class DividerWidgetComponent extends Widget {
+
+  constructor(private wS: WidgetService) {
+    super(wS);
+  }
+
+  updateData(data: any[]) {
```
Comment (matthiasbalke) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts]: Remove unimplemented method, as this widget does not update any data and the default implementation does the same.

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+  }
+
+  ngOnInit() {
```
Comment (matthiasbalke) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts]: Remove unimplemented method, as it is not needed by your component.

File: doc/examples/divider/createWidgets.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+HOST=localhost
+PORT=8080
+ENDPOINT=http://$HOST:8080/api/dashboards/1/widgets
+
+curl -X POST -d @widget.divider.json $ENDPOINT --header "Content-Type:application/json"
```
Comment (matthiasbalke) [inline doc/examples/divider/createWidgets.sh]: add a newline at the end of a file

File: doc/examples/divider/widget.divider.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "title": "Jenkins-Widgets",
+  "description": "of all our Testplatforms",
+  "type": "divider"
+}
```
Comment (matthiasbalke) [inline doc/examples/divider/widget.divider.json]: add a newline at the end of a file

File: doc/examples/jenkins/createWidgets.sh [hunk 7]
```diff
@@ -6,8 +6,4 @@ ENDPOINT=http://$HOST:8080/api/dashboards/1/widgets
 
 curl -X POST -d @widget.jenkins-job-notbuilt.json $ENDPOINT --header "Content-Type:application/json"
 curl -X POST -d @widget.jenkins-job-disabled.json $ENDPOINT --header "Content-Type:application/json"
-curl -X POST -d @widget.jenkins-job-green.json    $ENDPOINT --header "Content-Type:application/json"
-curl -X POST -d @widget.jenkins-job-yellow.json   $ENDPOINT --header "Content-Type:application/json"
-curl -X POST -d @widget.jenkins-job-aborted.json  $ENDPOINT --header "Content-Type:application/json"
-curl -X POST -d @widget.jenkins-job-red.json      $ENDPOINT --header "Content-Type:application/json"
-
+curl -X POST -d @widget.jenkins-job-green.json    $ENDPOINT --header "Content-Type:application/json"
```
Comment (matthiasbalke) [inline doc/examples/jenkins/createWidgets.sh]: Why did you remove the jenkins example widgets? Please restore the example.

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html (hunk 2 above)
Comment (RudiKlassen) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html]: changed font-size to 16px. It looks a little bit bigger than normal Text now.

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html (hunk 1 above)
Comment (RudiKlassen) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html]: Fixed it an other way. Pls check out my next pr update.

Comment (RudiKlassen) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.html]: Fixed it with a padding element in h5

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts (hunk 3 above)
Comment (RudiKlassen) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts]: done

File: dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts (hunk 4 above)
Comment (RudiKlassen) [inline dashy/src/app/dashboard/widget/divider-widget/divider-widget.component.ts]: done

File: doc/examples/divider/widget.divider.json (hunk 6 above)
Comment (RudiKlassen) [inline doc/examples/divider/widget.divider.json]: done

File: doc/examples/jenkins/createWidgets.sh (hunk 7 above)
Comment (RudiKlassen) [inline doc/examples/jenkins/createWidgets.sh]: Deleted them accidentally. Reverted my change.

File: doc/examples/divider/createWidgets.sh (hunk 5 above)
Comment (RudiKlassen) [inline doc/examples/divider/createWidgets.sh]: done

**Comment by matthiasbalke** [comment]
divider
